manifests/klusterlet/klusterlet-work-deployment.yaml

```diff
@@ -20,10 +20,21 @@ spec:
       - name: klusterlet-manifestwork-agent
         image: {{ .WorkImage }}
         imagePullPolicy: IfNotPresent
+        env:
+          {{if .ClusterName}}
+          - name: CLUSTER_NAME
+            value: {{ .ClusterName }}
+          {{else}}
+          - name: CLUSTER_NAME
+            valueFrom:
+              secretKeyRef:
+                name: {{ .HubKubeConfigSecret }}
+                key: cluster-name
+          {{end}}
         args:
           - "/work"
           - "agent"
-          - "--spoke-cluster-name={{ .ClusterName }}"
+          - "--spoke-cluster-name=$(CLUSTER_NAME)"
```

Comment:
doing it like this means that when the cluster-name changes, you will not automatically trigger a deployment, right?

Reply:
The env field will be change based on clusterName, I think that will trigger deployment.

Reply:
The deployment will be applied when the `spec.clusterName` field of the `Klusterlet` CR changes. So, it will switch correctly between:

1. `spec.clusterName=a`
2. `spec.clusterName=b`
3. `spec.clusterName` is unset
4. `spec.clusterName=c`

However, I am wondering whether the value will be correct when the system (3) - since the value inject will be the prior value set in the kubeconfig _before the pod starts_. So, I would expect when going from `spec.clusterName=a` to `spec.clusterName` is unset, that you would be injected with `a` when the pod started after the spec field for the name was unset.

Does that make sense?

Reply:
yes, that makes sense. In this case, the registration agent will be set with unset cluster name, and generate a new cluster name, then saved to the secret. However, work agent is already started using the old cluster name. I think we have to reconcile again when the cluster name in secret is changed. So we cannot use env var to ref secret here.